### ISSUE TITLE
BOP biomes not using BOP surface blocks?

### ISSUE THREAD

**Issue by whichonespink44** [Team-RTG/Realistic-Terrain-Generation#491]
http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/2524489-realistic-terrain-generation-rtg-realistic-biomes?comment=1560

I've checked a couple biomes and it looks like they *should* be using the BOP blocks (topBlock and fillerBlock are coded to use BOP's topBlock and fillerBlock), but haven't confirmed what's actually happening in-game as yet.

**Comment by srs-bsns** [comment]
Here's an example.

World type RTG:
![](http://www.nihilism.ca/minecraft/20160208-RTG-Issue491/20160208-Issue491-RTG.png)
World type BoP:
![](http://www.nihilism.ca/minecraft/20160208-RTG-Issue491/20160208-Issue491-BoP.png)

**Comment by Zeno410** [comment]
I remember there's a name conflict where there's more than one topBlock at different levels in the inheritance tree.

**Comment by srs-bsns** [comment]
Well, there's something that's taking precedence. I even tried using BiomeTweaker to force topBlock with:
```
.set("topBlock", "BiomesOPlenty:newBopGrass")
.set("topBlockMeta", 0)
```
It doesn't work.

**Comment by whichonespink44** [comment]
I'll be looking at this shortly. If anyone can think of any other BOP biomes that use BOP top blocks, just let me know and I'll see if I can sort those out as well. I know Origin Valley uses really bright green grass, but I think that one's working.

**Comment by Zeno410** [comment]
The *grass* is OK, the problem is the dirt.

**Comment by whichonespink44** [comment]
Fixed in 4b6d797 - The colour of the grass was correct, but it was using vanilla grass, which caused the rest of the texture to be brighter than OW's grass. To fix it, I just set the top and filler blocks to be BOP's 'new grass' and 'new dirt', which seems to have fixed it.

Still not sure why it wasn't honouring the biome's top & filler blocks, but maybe it's that name conflict bug you mentioned. Either way, Ominous Woods is fixed. Just need to find a resource where I can see which other biomes use BOP surface blocks. I suppose BOP's GitHub would be the best bet.

**Comment by srs-bsns** [comment]
According to the dump files that BiomeTweaker creates, these biomes all had the correct top and filler blocks set:

Biome | Top Block | Filler Block
------------ | ------------------------- | -------------
Canyon, Canyon Ravine|BiomesOPlenty:hardDirt|BiomesOPlenty:hardDirt
Crag	|BiomesOPlenty:cragRock|BiomesOPlenty:cragRock
Garden|BiomesOPlenty:longGrass|minecraft:dirt
Glacier, Polar Chasm|BiomesOPlenty:hardIce|BiomesOPlenty:hardIce
Origin Valley|BiomesOPlenty:originGrass|minecraft:dirt
Outback|BiomesOPlenty:hardSand	|BiomesOPlenty:hardSand
Quagmire|BiomesOPlenty:mud|BiomesOPlenty:mud
Volcano|BiomesOPlenty:ashStone|BiomesOPlenty:ashStone
Wasteland|BiomesOPlenty:driedDirt|BiomesOPlenty:driedDirt

But all of the BoP biomes (listed below) that use the top block: **BiomesOPlenty:newBopGrass**, and filler block: **BiomesOPlenty:newBopDirt** were instead listing minecraft:grass, and minecraft:dirt respectively.
```
Coniferous Forest
Snowwy Coniferous Forest
Dead Swamp
Moor
Ominous Woods
Sludgepit
Wetland
```

**Comment by whichonespink44** [comment]
Just getting back into this now. Thanks for the detailed info - that's a massive help.

**Comment by whichonespink44** [comment]
Fixed in 577936c and merged into `dev`
